unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneLifecycleHandler.cs

```diff
@@ -53,6 +53,19 @@ public SceneLifecycleHandler(ParcelScene ownerScene)
             sceneResourcesLoadTracker = new SceneResourcesLoadTracker();
             sceneResourcesLoadTracker.Track(owner.componentsManagerLegacy, Environment.i.world.state);
             sceneResourcesLoadTracker.OnResourcesStatusUpdate += OnResourcesStatusUpdated;
+   
+            // This is done while the two ECS are living together, if we detect that a component from the new ECS has incremented a 
+            // resource for the scene, we changed the track since that means that this scene is from the new ECS.
+            // This should disappear when the old ecs is removed from the project. This should be the default track 
+            DataStore.i.ecs7.pendingSceneResources.OnRefCountUpdated += ( sceneId,  amount) =>
+            {
+                if (sceneId.sceneId != ownerScene.sceneData.id)
+                    return;
+                
+                sceneResourcesLoadTracker.Dispose();
+                sceneResourcesLoadTracker.Track(sceneId.sceneId);
+                sceneResourcesLoadTracker.OnResourcesStatusUpdate += OnResourcesStatusUpdated;
+            };
```

Comment:
If we want to load a scene without pending resources, the scene will load normally with the old tracker because the `sceneResourceLoadTracker.ShouldWaitForPendingResources()`  will return false since there is no pending resources.

If no resources are never added, then we won't change the system loading tracker because it will work the same

It has more performance this way than doing everytime that a CRDT messages comes.

Reply:
yeah, but remember we want to delete the old ecs code in the future, so we can't depend on the other tracker.
you could use the event raised by `DataStore.i.ecs7.componentsManagers` since it would work just like this one but it won't depend on resources needed to be loaded